File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        Arrays.sort(happiness);
        int dec = 0;
        long max=0;
        int value=0;
        for(int i=happiness.length-1;i>=happiness.length-k;i--){
            value = happiness[i]-dec;
            dec++;
            if(value>0)
            {
                max += value;
            }
        }
        return max;
    }
}

File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.cpp
class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) {
        sort(happiness.rbegin(), happiness.rend());

        long long ans = 0;

        int cnt = 0;

        for(int it : happiness){
            if(it - cnt <= 0) break;
            
            ans += (it-cnt);
            cnt++;

            if(cnt == k) break;
        }

        return ans;
    }
};